### PR TITLE
LITTL-129 - Add prefix field to GuestTeacher

### PR DESCRIPTION
**Pull request by roelandkulk**

### CODE REVIEW

**Review by pschildkamp — CHANGES_REQUESTED** [review]
Could you add a test where we add a new guest teacher (with also a prefix) which is successfully persisted? Like the one you altered but then fetch the name and check if the added entity equals the properties of the DTO.

File: src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java [hunk 1]
```diff
@@ -1,52 +1,74 @@
...
+import java.util.*;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 
 @QuarkusTest
-@Disabled("Disabled, needs a lot of refactoring")
 class GuestGuestTeacherRepositoryTest {
```
Comment (pschildkamp) [inline src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java:20]: Twice Guest in the name.

File: src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java [hunk 2]
```diff
@@ -1,52 +1,74 @@
...
+
+        doReturn(List.of(expectedTeacher)).when(repository).list("surname like ?1", "%" + surname + "%");
+
+        List<GuestTeacherEntity> foundTeacherList = repository.findBySurnameLike(surname);
+
+        assertThat(foundTeacherList).isNotEmpty();
+
+        GuestTeacherEntity foundTeacher = foundTeacherList.get(0);
+
+        assertThat(expectedTeacher.getFirstName()).isEqualTo(foundTeacher.getFirstName());
```
Comment (pschildkamp) [inline src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java:67]: This is correct, but you should checkout https://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#field-by-field. That's a cool feature that makes you write less code ;). Let me know if you want to implement this or just go with what you got now.

**Review by pschildkamp — CHANGES_REQUESTED** [review]

File: src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java (hunk 1 above)
Comment (roelandkulk) [reply src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java]: Fixed

File: src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java (hunk 2 above)
Comment (roelandkulk) [reply src/test/java/org/littil/api/guestTeacher/repository/GuestGuestTeacherRepositoryTest.java]: Nice tip! Although these methods are deprecated now. It's replaced by usingRecursiveComparison().

**Review by pschildkamp — APPROVED** [review]
Great work.

**Pull request merged**